public/app/plugins/datasource/prometheus/components/PromLink.tsx

```diff
@@ -36,23 +35,17 @@ export default class PromLink extends Component<Props, State> {
       return '';
     }
 
-    const target = request.targets.length > 0 ? request.targets[0] : ({ datasource: null } as any);
-    const datasourceName = target.datasource;
-    const datasource: PrometheusDatasource = datasourceName
-      ? (((await getDatasourceSrv().get(datasourceName)) as any) as PrometheusDatasource)
-      : (this.props.datasource as PrometheusDatasource);
-
     const range = request.range;
-    const start = datasource.getPrometheusTime(range.from, false);
-    const end = datasource.getPrometheusTime(range.to, true);
+    const start = this.props.datasource.getPrometheusTime(range.from, false);
+    const end = this.props.datasource.getPrometheusTime(range.to, true);
```

Comment:
Any reason why we are not destructuring this.props so we could to just `datasource.getPrometheusTime(range.from, false);`? We are already doing ` const { query } = this.props;`.

Reply:
Good finding! Let me destruct datasource then.